spring-integration-core/src/main/java/org/springframework/integration/config/annotation/InboundChannelAdapterAnnotationPostProcessor.java

```diff
@@ -72,7 +74,16 @@ public Object postProcess(Object bean, String beanName, Method method, List<Anno
 			return null;
 		}
 
-		MessageChannel channel = this.channelResolver.resolveDestination(channelName);
+		MessageChannel channel;
+		try {
+			channel = this.channelResolver.resolveDestination(channelName);
+		}
+		catch (DestinationResolutionException e) {
+			channel = new DirectChannel();
+			this.beanFactory.registerSingleton(channelName, channel);
+			channel = (MessageChannel) this.beanFactory.initializeBean(channel, channelName);
+		}
+		Assert.notNull(channel, "failed to resolve channel '" + channelName + "'");
```

Comment:
Yes, that sounds better. That's why all those my expression here around this fix are not so 	certain, because I'm not sure in the fix, too.
E.g. with DSL we can do 
```
IntegrationFlows.from(c -> c.queue("myQueueChannel"))
```
When we register that channel from the `BPP` anyway.
So, `InboundChannelAdapterAnnotationPostProcessor` may be initiated early and we won't have the channel yet.
Therefore something like Late Binding would be better.

Your turn!